### PR TITLE
Remove direct access to small scenery's colour_1 and _2 fields

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
The functions should really be called `primary` and `secondary` not `colour 1` and `colour 2`. The reason they are called `colour 1` and `2` is because there is more data in there than just primary and secondary.

File: src/openrct2/paint/map_element/scenery.c [hunk 1]
```diff
@@ -254,7 +263,7 @@ void scenery_paint(uint8 direction, sint32 height, rct_map_element* mapElement)
         }
     }
     // 6E0556: Draw supports:
-    if (mapElement->properties.scenery.colour_1 & 0x20) {
+    if (scenery_get_colour_1(mapElement) & 0x20) {
```
Comment (duncanspumpkin) [inline src/openrct2/paint/map_element/scenery.c]: Mistake. This is getting 0x1F and anding with 0x20

File: src/openrct2/world/scenery.c [hunk 2]
```diff
@@ -317,3 +317,33 @@ sint32 get_scenery_id_from_entry_index(uint8 objectType, sint32 entryIndex)
...
+void scenery_set_colour_2(rct_map_element * mapElement, uint32 colour)
+{
+    assert(colour <= 31);
+    mapElement->properties.scenery.colour_2 &= ~0x1F;
+    mapElement->properties.scenery.colour_2 |= colour;
+}
+
+void scenery_set_supports_needed(rct_map_element * mapElement)
+{
+    mapElement->properties.scenery.colour_1 |= (1 << 5);
```
Comment (duncanspumpkin) [inline src/openrct2/world/scenery.c]: Define this flag.

**Comment by Gymnasiast** [comment]
So I should rename the functions to `scenery_get_primary_colour` and `scenery_get_secondary_colour`?

**Comment by duncanspumpkin** [comment]
Yes in general I've used primary and secondary to refer to the colours but I'm not sure if the rest of the code base has followed that convention.

File: src/openrct2/world/scenery.h [hunk 3]
```diff
@@ -80,6 +80,10 @@ typedef enum {
     SMALL_SCENERY_FLAG27 = (1 << 27),                           // 0x8000000
 } SMALL_SCENERY_FLAGS;
 
+typedef enum {
```
Comment (duncanspumpkin) [inline src/openrct2/world/scenery.h]: This is a map element small scenery flag the union is defined in the map_element file. Should probably move this over to there. Also its not clear that this is for small scenery map elements

File: src/openrct2/rct1/S4Importer.cpp [hunk 4]
```diff
@@ -2198,8 +2198,8 @@ class S4Importer final : public IParkImporter
             {
                 switch (map_element_get_type(mapElement)) {
                 case MAP_ELEMENT_TYPE_SCENERY:
-                    colour = RCT1::GetColour(scenery_get_colour_1(mapElement));
-                    scenery_set_colour_1(mapElement, colour);
+                    colour = RCT1::GetColour(scenery_get_primary_colour(mapElement));
+                        scenery_set_primary_colour(mapElement, colour);
```
Comment (duncanspumpkin) [inline src/openrct2/rct1/S4Importer.cpp]: some stray space got in.

**Pull request merged**